addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1348,27 +1348,26 @@ const Wysiwyg = Widget.extend({
      * @param {Node} [params.htmlClass] Optionnal
      */
     openMediaDialog(params = {}) {
-        const sel = this.odooEditor.document.getSelection();
-
-        if (!sel.rangeCount) {
-            return;
-        }
-        const range = sel.getRangeAt(0);
         // We lose the current selection inside the content editable when we
         // click the media dialog button so we need to be able to restore the
         // selection when the modal is closed.
         const restoreSelection = preserveCursor(this.odooEditor.document);
 
-        const $editable = $(OdooEditorLib.closestElement(range.startContainer, '.o_editable') || this.odooEditor.editable);
-        const model = $editable.data('oe-model');
-        const field = $editable.data('oe-field');
-        const type = $editable.data('oe-type');
+        const $el = $(params.node || this.odooEditor.document.getSelection().anchorNode);
+        const $closestBrandedEditable = $el.closest('.o_editable:data(oe-model):data(oe-id),.o_editable:data(oe-media-domain)');
+        const $editable = $closestBrandedEditable[0] || $(this.odooEditor.editable);
+        
+        const resModel =  $editable.data('oe-model') || 'ir.attachment';
+        const resId =  $editable.data('oe-id') || false;
+        const domain =  $editable.data('oe-media-domain') || [];
+        const type =  $editable.data('oe-type') || false;
+        const field =  $editable.data('oe-field') || false;
```

Comment:
I don't understand why we would want to replace `undefined` with `false`. In practice it probably work because the function later on probably doesn't distinguish between `false` and `undefined`, but it is conceptually wrong IMO. This is more visible if we imagine it from a typing perspective. In Typescript for example, we would type the options like `{ resId?: <type of resId>, ... }`, but by saying `false` is a possible value, you'd now have something like `{ resId?: <type of resId> | false, ... }` and it just seems superfluous considering `resId` can already be `undefined` through the use of the `?` modifier. This is less visible for `domain` for which the default value of `[]` matches its own type, but it seems superfluous nonetheless because the code that handles those options probably handles the case where no domain is given since `domain` is optional, and an omitted  parameter in JS _is_ the value `undefined`, so I don't see the point to replace those by arbitrary default values at this point.

Reply:
You are right; undefined is fine. I just used false because it felt natural since the value where then used in a boolean expression. I removed the fallback value.